expression/builtin.go

```diff
@@ -355,3 +357,35 @@ func builtinLeast(args []types.Datum, ctx context.Context) (d types.Datum, err e
 	d = args[min]
 	return
 }
+
+// See http://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_interval
+func builtinInterval(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
+	if args[0].IsNull() {
+		d.SetInt64(int64(-1))
+		return
+	}
+	sc := ctx.GetSessionVars().StmtCtx
+
+	lt := func(d1, d2 types.Datum) bool {
+		if d1.Kind() == types.KindString || d2.Kind() == types.KindString {
```

Comment:
I don't think we need take much consideration for this. 
user will get wrong result because their input is illegal.
when the N1...Nn is not sorted, the result is also undefined since we use binary search.